src/aws/nodemanager/nodemanager.py

```diff
@@ -92,6 +97,27 @@ def add_task(self, task):
         """
         raise NotImplementedError()
 
+    def steal_task(self):
+        """Steals a task from the worker with the most number of tasks"""
+        assignments = []
+        for worker in self._task_assignment:
+            assignments.append(len(self._task_assignment[worker]))
+        victim_worker = list(self._task_assignment.keys())[assignments.index(max(assignments))]
+        stolen_task = self._task_assignment[victim_worker].pop(0)
+        return stolen_task
```

Comment:
You could replace this with `return self._task_assignment[victim_worker].pop(0)`